src/main/python/main/ayab/engine/communication_ip.py

```
@@ -0,0 +1,215 @@
+    def open_tcp(self, pPortname=None):
+        try:
+            self.__portname = pPortname
+            self.__tarAddressPort = (self.__portname, remotePort)
+            self.__sockTCP = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+            self.__sockTCP.settimeout(10.0)
+            self.__sockTCP.connect(self.__tarAddressPort)
+            self.__sockTCP.settimeout(0.0)
+            self.__sockTCP.setblocking(False)
+            self.logger.info("Open TCP Socket successful")
+            return True
+        except:
```

Ruff

Do not use bare `except`

(E722)